db/warehouse/migrate/20240430045111_custom_service_type_slug.rb

```diff
@@ -0,0 +1,25 @@
+class CustomServiceTypeSlug < ActiveRecord::Migration[6.1]
+  def up
+    safety_assured do
+      execute <<~SQL
+        ALTER TABLE "Services" ADD COLUMN custom_service_type_slug VARCHAR GENERATED ALWAYS AS (data_source_id::text || ':' || "TypeProvided"::text || ':' || "RecordType"::text) STORED
+      SQL
+      execute <<~SQL
+        ALTER TABLE "CustomServiceTypes" ADD COLUMN slug VARCHAR GENERATED ALWAYS AS (data_source_id::text || ':' || "hud_type_provided"::text ||':'|| "hud_record_type"::text) STORED
+      SQL
+      execute %{CREATE INDEX idx_custom_service_type_slug ON "CustomServiceTypes" (slug)}
+      execute %(VACUUM ANALYZE "Enrollment")
+      execute %(VACUUM ANALYZE "CustomServices")
+      execute %(VACUUM ANALYZE "Services")
+    end
+  end
+
+  def down
+    safety_assured do
+      change_column_null :Services, :EnrollmentID, true
+      change_column_null :Services, :PersonalID, true
```

Comment:
What's this for? I don't see anywhere we make them required on `up`

Reply:
Ah oops, I forgot to take this out. I had initially changed the nullability of Services to match CustomServices but I don't think that matters for performance